### PR TITLE
Constants are non empty strings

### LINKED ISSUE #6266: non empty string constants should be considered as a non empty string
With the option
```
ignoreInternalFunctionFalseReturn="false"
```

In
```
/** @var non-empty-string $foo */
$foo = PHP_EOL;
$bar = explode($foo, $diff);
```
$bar is considered as always an array.

In
```
$bar = explode(PHP_EOL, $diff);
```
$bar is considered as array|false.

So I would say that explode is correctly supporting non-empty-string, but some constant are not considered as non-empty-string. I tried
- PHP_EOL
- DIRECTORY_SEPARATOR
- a custom class constant `public const FOO = 'foo';`

Should these be `non-empty-string` ?
https://psalm.dev/r/377464f1cd

### PR DESCRIPTION
**Pull request by VincentLanglet**
Close #6266

### CODE REVIEW

File: src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php [hunk 1]
```diff
@@ -155,7 +155,7 @@ public static function getGlobalConstType(
                 case 'PHP_PREFIX':
                 case 'PHP_SAPI':
                 case 'PHP_SYSCONFDIR':
-                    return Type::getString();
```
Comment (weirdan) [inline src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php:158]: Not all of those are necessarily non-empty: https://3v4l.org/oSTv2

Comment (weirdan) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php:158]: I would suggest to limit the change to those constants you had in your original issue.

Comment (VincentLanglet) [reply src/Psalm/Internal/Analyzer/Statements/Expression/Fetch/ConstFetchAnalyzer.php:158]: I found 3 constants possibly empty and fixed them.

**Comment by VincentLanglet** [comment]
@weirdan I followed your advice and only return nonEmptyString for the most useful
PHP_VERSION, DIRECTORY_SEPARATOR, PATH_SEPARATOR and PHP_EOL.

**Review by weirdan — APPROVED** [review]

**Comment by weirdan** [comment]
Thanks!

**Pull request merged**